### PR TITLE
users can sheet and view a list of their own sheets

### PR DESCRIPTION
**Pull request by robertpulson**

### CODE REVIEW

File: spec/features/showing_all_tweets_spec.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  }
+
+  scenario 'when visiting the homepage' do
+    visit '/'
+    expect(page).to have_content('This is a Sheet!')
+  end
+  
+  scenario 'ordered by time' do
+    visit '/'
+    page.body.index('This is also a Sheet!').should < page.body.index('This is a Sheet!')
```
Comment (tansaku) [inline spec/features/showing_all_tweets_spec.rb]: you should use the new rspec 'expect' syntax - we challenge you to re-write this with 'expect'

File: spec/features/showing_all_tweets_spec.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+require 'spec_helper'
+
+feature 'User browses the list of tweets' do
+
+  before(:each) {
```
Comment (tansaku) [inline spec/features/showing_all_tweets_spec.rb]: when multi-line a block should be 'do ... end' and not '{}'

File: spec/features/step_definitions/web_steps.rb [hunk 3]
```diff
@@ -0,0 +1,219 @@
+# Taken from the cucumber-rails project.
```
Comment (tansaku) [inline spec/features/step_definitions/web_steps.rb]: you aren't using cucumber, don't include files like this that are not necessary (they do more harm than good) - note also that your directory structure here is strange.  The contents of step_definitions and support directories are completely redundant and should be removed.

File: app/server.rb [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  erb :index
+end
+
+get '/users/new' do
+  @user = User.new
+  erb :'users/new'
+end
+
+post '/users' do
+  @user = User.create(:name => params[:name],
```
Comment (tansaku) [inline app/server.rb]: adjusting params could be extracted to a separate method to reduce the size of this one

File: app/server.rb [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  session[:user_id] = nil
+  redirect to('/')
+end
+
+post '/sheets/new' do
+  erb :'sheets/new'
+end
+
+post '/sheet' do
+   Sheet.create(:text => params[:sheet],
```
Comment (tansaku) [inline app/server.rb]: indenting is incorrect here - 3 spaces in bad

File: app/server.rb [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+DataMapper.auto_upgrade!
+
+enable :sessions
+set :session_secret, 'super secret'
+use Rack::Flash
+use Rack::MethodOverride
+
+set :public_folder, Proc.new { File.join(root, '..', "public") }
+
+get '/' do
```
Comment (tansaku) [inline app/server.rb]: we recommend using Sinatra::Application here

File: app/server.rb [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+post '/user/profile' do
+  @sheets = Sheet.all(:user_id => session[:user_id])
+  erb :'users/profile'
+end
+
+helpers do
+
+  def current_user
+    @current_user ||=User.get(session[:user_id]) if session[:user_id]
```
Comment (tansaku) [inline app/server.rb]: ensure you have a space after '||='

File: spec/sheet_spec.rb [hunk 8]
```diff
@@ -0,0 +1,7 @@
+require 'spec_helper'
```
Comment (tansaku) [inline spec/sheet_spec.rb]: this file is not really doing anything - so you should remove it, or add unit tests to it.

File: app/views/layout.erb [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+    <title>5hitter</title>
+    <link rel='stylesheet' type='text/css' href='style.css'>
+  </head>
+
+  <body>
+
+    <h1>5hitter</h1>
+  
+    <% if flash[:notice] %>
+      <div id='notice'>
```
Comment (tansaku) [inline app/views/layout.erb]: don't use divs

File: app/helpers/session [hunk 10]
```diff
@@ -0,0 +1,6 @@
+    def sign_in(username, password)
```
Comment (tansaku) [inline app/helpers/session]: this seems like it's not doing anything - please delete

**Comment by spike01** [comment]
:checkered_flag:

**Comment by tansaku** [comment]
Great to see it all working including sign up and log in etc.  Functionality is all there.  Please just ensure to remove any extraneous code and files that are not being used.  Well done for completing the challenge - you've earned the 'Data' badge.

File: app/server.rb [hunk 11]
```diff
@@ -0,0 +1,90 @@
+require 'sinatra'
+require 'data_mapper'
+require 'rack-flash'
+
+env = ENV['RACK_ENV'] || 'development'
```
Comment (tansaku) [inline app/server.rb]: consider extracting all the database setup to a separate file

**Pull request closed without merging**